src/preferences/game.cpp

```diff
@@ -230,6 +230,23 @@ const std::map<std::string, acquaintance> & get_acquaintances() {
 	return acquaintances;
 }
 
+const std::string get_ignored_delim() {
+	std::string ignored = "";
+	load_acquaintances();
+	
+	for(std::map<std::string, acquaintance>::iterator i = acquaintances.begin(); i != acquaintances.end(); ++i)
+	{
+		if(i->second.get_status() == "ignore") {
+			if(ignored != "") {
+				ignored.append(",");
+			}
+			ignored.append(i->second.get_nick());
+		}
+	}
+	
+	return ignored;
+}
+
```

Comment:
Cleaner impl (untested):
```cpp
const std::string get_ignored_delim()
{
    load_acquaintances();
    std::vector<std::string> ignored;

    for(const auto& person : acquaintances) {
        if(person.second.get_status() == "ignore") {
            ignored.push_back(person.second.get_nick())
        }
    }

    return utils::join(ignored);
}
```

Reply:
Updated and tested.

Reply:
Or even (also untested)

    const std::string get_ignored_delim()
    {
        load_acquaintances();
        std::vector<std::string> ignored;
    
        std::copy_if(acquaintances.begin()
            , acquaintances.end()
            , std::back_inserter(ignored)
            , [](const auto& person){
                return person.second.get_status() == "ignore";
            }
        );
    
        return utils::join(ignored);
    }

if you want to fully get rid of manual iteration.

Reply:
@Vultraz is there a preference between which way is used in Wesnoth's codebase?

Reply:
Not particularly. Second is a bit more compact though, so I might go with that myself, but it doesn't really matter.

Reply:
Alright.  In any case, it turns out the above doesn't work, since it can't copy the `pair<std::string, acquaintance>` element into the `std::vector<std::string>`.